libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -316,6 +316,10 @@ void AP_MotorsHeli_Dual::mix_tandem(float pitch_input, float roll_input, float y
     const float swash1_coll =  0.45 * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective1_input;
     const float swash2_coll = -0.45 * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective2_input;
 
+    // Log blade pitch angle contributions
+    log_swashplate(0, swash1_roll, swash_pitch, swash1_coll, 1.0);
+    log_swashplate(1, swash2_roll, swash_pitch, swash2_coll, 1.0);
```

Comment:
Move "memory" portion down in to the swash plate object. You would have to add the cyclic scale as constant passed in the constructor. The you can do the "calculate" part then you log.